src/StatusQ/Controls/StatusInput.qml

```diff
@@ -348,7 +348,9 @@ Item {
     }
 
     implicitWidth: 448
-    implicitHeight: (internal.inputHeight + topRow.height + errorMessage.height + (2*inputLayout.spacing))
+    implicitHeight: (internal.inputHeight + topRow.height + errorMessage.height +
+                    ((topRow.height > 0) && (errorMessage.height > 0) ? 16 :
+                    (topRow.height > 0) || (errorMessage.height > 0) ? 8 : 0))
```

Comment:
This is becoming too much... can't we do some `layout.childrenRect` instead to simplify this?

Reply:
it didn't work out of the box, at least as a quick 'fix', the layout was not quite reacting to height changes of its children, the 'spaghetti' is just for the spacing to be taken into account.